[BUG] The ci docker images fail to build with a gradle error

**Describe the bug**
Trying to run `docker/ci/ci.sh up` results in the images failing to complete the `gradle clean` stage with the following error:

```
 > [wi-build builder 4/4] RUN gradle clean:                                                                                                                                           
0.749 Starting a Gradle Daemon (subsequent builds will be faster)                                                                                                                     
13.55 > Task :buildSrc:reaper:compileJava                                                                                                                                             
13.55 > Task :buildSrc:reaper:processResources NO-SOURCE                                                                                                                              
13.55 > Task :buildSrc:reaper:classes                                                                                                                                                 
13.55 > Task :buildSrc:reaper:jar                                                                                                                                                     
26.85                                                                                                                                                                                 
26.85 > Task :buildSrc:compileJava                                                                                                                                                    
26.85 /home/wazuh-indexer/buildSrc/src/main/java/org/opensearch/gradle/info/GlobalBuildInfoPlugin.java:202: error: constructor InstallationLocation in class InstallationLocation cannot be applied to given types;                                                                                                                                                         
26.85         final InstallationLocation location = new InstallationLocation(javaHome, "Java home");                                                                                  
26.85                                               ^
26.85   required: File,String,boolean,boolean
26.85   found:    File,String
26.85   reason: actual and formal argument lists differ in length
26.85 Note: Some input files use or override a deprecated API.
26.85 Note: Recompile with -Xlint:deprecation for details.
26.85 1 error
26.95 
26.95 > Task :buildSrc:compileJava FAILED
26.95 
26.95 FAILURE: Build failed with an exception.
26.95 
26.95 * What went wrong:
26.95 Execution failed for task ':buildSrc:compileJava'.
26.95 > Compilation failed; see the compiler error output for details.
26.95 
26.95 * Try:
26.95 > Run with --info option to get more log output.
26.95 
26.95 BUILD FAILED in 26s
26.95 3 actionable tasks: 3 executed
```

**To Reproduce**
* Clone the indexer repo and switch to it
* Go into `docker/ci`
* Run `ci.sh up`

**Expected behavior**
Docker images should be built and containers brought up that should be capable of building our packages.

--- a/docker/dev/images/Dockerfile
+++ b/docker/dev/images/Dockerfile
@@ -1,4 +1,4 @@
-FROM gradle:jdk21-alpine AS builder
+FROM gradle:8.7.0-jdk21-alpine AS builder
 USER gradle
 WORKDIR /home/wazuh-indexer
 COPY --chown=gradle:gradle . /home/wazuh-indexer

